presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -312,4 +326,22 @@ private void enforceCpuLimits()
             }
         }
     }
+
+    /**
+     * Enforce query scan physical bytes limits
+     */
+    private void enforceScanLimits()
+    {
+        for (QueryExecution query : queryTracker.getAllQueries()) {
+            DataSize limitOpt = Optional.ofNullable(getQueryMaxScanPhysicalBytes(query.getSession()))
```

Comment:
`getQueryMaxScanPhysicalBytes(session)` returns a value that can be used as is without further checking.  

This code ensures that this property defaults to what's specified in config:

```
                dataSizeProperty(
                        QUERY_MAX_SCAN_PHYSICAL_BYTES,
                        "Maximum scan physical bytes of a query",
                        queryManagerConfig.getQueryMaxScanPhysicalBytes(),
                        false),
```

and this code defines the default value that is used if nothing is specified in config file:

```
private DataSize queryMaxScanPhysicalBytes = DataSize.succinctDataSize(1, PETABYTE);
```

Hence, the code can be simplified like this:

```
        for (QueryExecution query : queryTracker.getAllQueries()) {
            DataSize limit = getQueryMaxScanPhysicalBytes(query.getSession());
            DataSize scan = query.getQueryInfo().getQueryStats().getRawInputDataSize();
            if (scan.compareTo(limit) >= 0) {
                query.fail(new ExceededScanLimitException(limit));
            }
        }
```

Reply:
Many thanks for the proposal. I have made another version of how to implement this limit based on your suggestion. The new version is now includes a method defines in QueryExecution to collect scanned data size which is implemented in SqlQueryExecution to collect finalQueryInfo's rawInputDataSize. Please help review again.